pocs/utils/logger.py

```diff
@@ -9,47 +9,30 @@
 
 from pocs.utils.config import load_config
 
+# We don't want to create multiple root loggers that are "identical",
+# so track the loggers in a dict keyed by a tuple of:
+#    (profile, json_serialized_logger_config).
+all_loggers = {}
 
-class PanLogger(object):
-    """ Logger for PANOPTES with format style strings """
-
-    def __init__(self, logger):
-        super(PanLogger, self).__init__()
-        self.logger = logger
-
-    def _process_str(self, fmt, *args, **kwargs):
-        """ Pre-process the log string
-
-        This allows for `format` style specifiers, e.g. `{:02f}` and
-        `{:40s}`, which otherwise aren't supported by python's default
-        log formatting.
-        """
-        log_str = fmt
-        if len(args) > 0 or len(kwargs) > 0:
-            log_str = fmt.format(*args, **kwargs)
-
-        return log_str
-
-    def debug(self, fmt, *args, **kwargs):
-        self.logger.debug(self._process_str(fmt, *args, **kwargs))
-
-    def info(self, fmt, *args, **kwargs):
-        self.logger.info(self._process_str(fmt, *args, **kwargs))
-
-    def warning(self, fmt, *args, **kwargs):
-        self.logger.warning(self._process_str(fmt, *args, **kwargs))
 
-    def error(self, fmt, *args, **kwargs):
-        self.logger.error(self._process_str(fmt, *args, **kwargs))
+class StrFormatLogRecord(logging.LogRecord):
+    """ Allow for `str.format` style log messages
 
-    def critical(self, fmt, *args, **kwargs):
-        self.logger.critical(self._process_str(fmt, *args, **kwargs))
+    Even though you can select '{' as the style for the formatter class,
+    you still can't use {} formatting for your message. The custom
+    `getMessage` tries legacy format and then tries new format.
 
+    From: https://goo.gl/Cyt5NH
+    """
 
-# We don't want to create multiple root loggers that are "identical",
-# so track the loggers in a dict keyed by a tuple of:
-#    (profile, json_serialized_logger_config).
-all_loggers = {}
+    def getMessage(self):
+        msg = str(self.msg)
+        if self.args:
+            try:
+                msg = msg % self.args
+            except (TypeError, ValueError):
+                msg = msg.format(*self.args)
+        return msg
 
 
 def get_root_logger(profile='panoptes', log_config=None):
```

Comment:
Please document params, especially point to https://docs.python.org/3/library/logging.config.html#configuration-dictionary-schema to explain log_config.